pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -37,7 +37,7 @@ type SDFile string
 // NamespaceDiscovery is the configuration for discovering
 // Kubernetes namespaces.
 type NamespaceDiscovery struct {
-	// Includes the namespace in which the Prometheus pod exists to the list of watched namesapces.
+	// Includes the namespace in which the Prometheus pod exists, to the list of watched namespaces.
```

Comment:
btw we could also hardcode the namespace in the generated config instead of setting `own_namespace: true` since the operator always knows the namespace.
```suggestion
	// Includes the namespace in which the Prometheus pod runs to the list of watched namespaces.
```

Reply:
Hmm... I get what you are saying but I do not see why we may want to do that. Letting the user specify it is more explicit IMO: the user knows what to expect from the generated config in that case. It might confuse the user to see that `own_namespace: true` is not present but the behavior is as expected.

Reply:
> but I do not see why we may want to do that.

"own_namespace: true" tells the Prometheus container to read the namespace value from `/var/run/secrets/kubernetes.io/serviceaccount/namespace`. There's an edge case where the read operation fails and the service discovery returns no targets which would be impossible if we hardcode the namespace value. Again this isn't for this PR and probably not a change worth spending time.

Reply:
I see, you mean to keep `own_namespace:true` and also adding it to the list of namespaces for good measure. That sounds like a good idea. That being said, I do agree we can do it in a later PR if we'd like.